src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionId = (data);
-            partitionId = Utils::trim_copy(partitionId, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID: " + partitionId, "info");
+    if (response.compare(JasmineGraphInstanceProtocol::SEND_FILE) == 0) {
+        std::cout << "Sending file " << filePath << " through port " << dataPort << std::endl;
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionIdList = (data);
-            partitionIdList = Utils::trim_copy(partitionIdList, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID List : " + partitionIdList, "info");
+        FILE *fp = fopen(filePath.c_str(), "r");
+        if (fp == NULL) {
+            instance_logger.log("Error opening file", "error");
+            close(sockfd);
+            return false;
+        }
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string priority = (data);
-            priority = Utils::trim_copy(priority, " \f\n\r\t\v");
-            instance_logger.log("Received priority: " + priority, "info");
+        for (;;) {
+            unsigned char buff[INSTANCE_FILE_BUFFER_LENGTH] = {0};
+            int nread = fread(buff, 1, INSTANCE_FILE_BUFFER_LENGTH, fp);
+            instance_logger.log("Bytes read " + to_string(nread), "info");
 
-            int threadPriority = std::atoi(priority.c_str());
+            /* If read was success, send data. */
+            if (nread > 0) {
+                instance_logger.log("Sending", "info");
 
-            if (threadPriority > Conts::DEFAULT_THREAD_PRIORITY) {
-                threadPriorityMutex.lock();
-                workerHighPriorityTaskCount++;
-                highestPriority = threadPriority;
-                threadPriorityMutex.unlock();
+                write(sockfd, buff, nread);
             }
 
-            std::string aggregatedTriangles = JasmineGraphInstanceService::aggregateCentralStoreTriangles(
-                graphId, partitionId, partitionIdList, threadPriority);
-
-            if (threadPriority > Conts::DEFAULT_THREAD_PRIORITY) {
-                threadPriorityMutex.lock();
-                workerHighPriorityTaskCount--;
-
-                if (workerHighPriorityTaskCount == 0) {
-                    highestPriority = Conts::DEFAULT_THREAD_PRIORITY;
-                }
-                threadPriorityMutex.unlock();
+            if (nread < INSTANCE_FILE_BUFFER_LENGTH) {
+                if (feof(fp)) printf("End of file\n");
```

Comment:
Rather than having `printf()` can't we use logger here? Also lets change single line `if` statements to have curly braces like ```if() {

}```

Reply:
This comments applies to both the lines having `if`

Reply:
Updated in next PR

Reply:
Ok